internal/assistedserviceiso/assistedserviceiso_api.go

```diff
@@ -0,0 +1,132 @@
+		}
+	} else {
+		log.Warn("Pull-secret for live ISO must be provided")
+		return assisted_service_iso.NewCreateISOAndUploadToS3BadRequest().
+			WithPayload(common.GenerateError(http.StatusBadRequest, errors.New("Pull-secret must be provided for live ISO")))
+	}
+
+	data, err := ioutil.ReadFile(a.ignitionConfigBaseFilename)
+	if err != nil {
+		log.WithError(err).Errorf("Error reading ignition config file")
+		return assisted_service_iso.NewCreateISOAndUploadToS3InternalServerError().WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+	ignitionConfigSource := string(data)
+
+	reIgnition := strings.NewReplacer("replace-with-your-ssh-public-key", sshPublicKey,
+		"replace-with-your-urlencoded-pull-secret", url.PathEscape(pullSecret))
+
+	ignitionConfig := reIgnition.Replace(ignitionConfigSource)
+
+	username := auth.UserNameFromContext(ctx)
+	isoName := imgexpirer.AssistedServiceLiveISOPrefix + username
+
+	if err = a.objectHandler.UploadISO(ctx, ignitionConfig, isoName); err != nil {
+		log.WithError(err).Errorf("Failed to generate live ISO")
+		return assisted_service_iso.NewCreateISOAndUploadToS3InternalServerError().WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+
+	return assisted_service_iso.NewCreateISOAndUploadToS3Created()
+}
+
+/* DownloadISO Downloads the ISO for the user from S3. */
+func (a *assistedServiceISOApi) DownloadISO(ctx context.Context, params assisted_service_iso.DownloadISOParams) middleware.Responder {
+	log := logutil.FromContext(ctx, a.log)
+
+	username := auth.UserNameFromContext(ctx)
+	isoName := imgexpirer.AssistedServiceLiveISOPrefix + username
+
+	reader, contentLength, err := a.objectHandler.Download(ctx, isoName+".iso")
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get Assisted Service ISO for user: %s", username)
+		return assisted_service_iso.NewDownloadISOInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+
+	return filemiddleware.NewResponder(assisted_service_iso.NewDownloadISOOK().WithPayload(reader),
+		isoName+".iso",
+		contentLength)
+}
+
+func (a *assistedServiceISOApi) GetPresignedForAssistedServiceISO(ctx context.Context, params assisted_service_iso.GetPresignedForAssistedServiceISOParams) middleware.Responder {
+	log := logutil.FromContext(ctx, a.log)
+	// Presigned URL only works with AWS S3 because Scality is not exposed
+	if !a.objectHandler.IsAwsS3() {
+		return common.NewApiError(http.StatusBadRequest, errors.New("Failed to generate presigned URL: invalid backend"))
+	}
+
+	username := auth.UserNameFromContext(ctx)
+	isoName := imgexpirer.AssistedServiceLiveISOPrefix + username
+	isoNameWithExtension := isoName + ".iso"
+
+	duration, _ := time.ParseDuration("10m")
+	url, err := a.objectHandler.GeneratePresignedDownloadURL(ctx, isoName, isoNameWithExtension, duration)
```

Comment:
Rather than setting to 10m, set to the image expiration time  - b.Config.ImageExpirationTime